app/src/main/java/com/stevesoltys/seedvault/storage/WebDavStoragePlugin.kt

```diff
@@ -0,0 +1,290 @@
+                ) { response, relation ->
+                    debugLog { "getAvailableChunkIds() = $response" }
+                    // This callback will be called for every file in the folder
+                    if (relation != SELF && response.isFolder()) {
+                        val name = response.hrefName()
+                        if (chunkFolderRegex.matches(name)) {
+                            expectedChunkFolders.remove(name)
+                        }
+                    } else if (relation != SELF && response.href.pathSize >= 2) {
+                        val folderName =
+                            response.href.pathSegments[response.href.pathSegments.size - 2]
+                        if (folderName != folder && chunkFolderRegex.matches(folderName)) {
+                            val name = response.hrefName()
+                            if (chunkRegex.matches(name)) chunkIds.add(name)
+                        }
+                    }
+                }
+            }
+            Log.i(TAG, "Retrieving chunks took $duration")
+        } catch (e: NotFoundException) {
+            debugLog { "Folder not found: $location" }
+            davCollection.createFolder()
+        } catch (e: Exception) {
+            if (e is IOException) throw e
+            else throw IOException("Error populating chunk folders: ", e)
+        }
+        Log.i(TAG, "Got ${chunkIds.size} available chunks")
+        createMissingChunkFolders(expectedChunkFolders)
+        return chunkIds
+    }
+
+    @Throws(IOException::class)
+    private suspend fun createMissingChunkFolders(
+        missingChunkFolders: Set<String>,
+    ) {
+        val s = missingChunkFolders.size
+        for ((i, chunkFolderName) in missingChunkFolders.withIndex()) {
+            val location = "$url/$folder/$chunkFolderName".toHttpUrl()
+            val davCollection = DavCollection(okHttpClient, location)
+            val response = davCollection.createFolder()
+            debugLog { "Created missing folder $chunkFolderName (${i + 1}/$s) $response" }
+        }
+    }
+
+    override fun getMasterKey(): SecretKey = keyManager.getMainKey()
+    override fun hasMasterKey(): Boolean = keyManager.hasMainKey()
+
+    @Throws(IOException::class)
+    override suspend fun getChunkOutputStream(chunkId: String): OutputStream {
+        val chunkFolderName = chunkId.substring(0, 2)
+        val location = "$url/$folder/$chunkFolderName/$chunkId".toHttpUrl()
+        debugLog { "getChunkOutputStream($location) for $chunkId" }
+        return try {
+            getOutputStream(location)
+        } catch (e: Exception) {
+            if (e is IOException) throw e
+            else throw IOException("Error getting OutputStream for $chunkId: ", e)
+        }
+    }
+
+    @Throws(IOException::class)
+    override suspend fun getBackupSnapshotOutputStream(timestamp: Long): OutputStream {
+        val location = "$url/$folder/$timestamp.SeedSnap".toHttpUrl()
```

Comment:
Are we using the file extension `SeedSnap`? Do we need this to be a constant to avoid mistakes?

Reply:
Alright, pushed a commit that does this project-wide.